src/index.js

```diff
@@ -1,32 +1,67 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const API = 'https://rickandmortyapi.com/api/character/';
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const fetchData = url_api => {
+  return new Promise((resolve, reject) => {
+    const xhttp = new XMLHttpRequest();
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
+    xhttp.onreadystatechange = event => {
+      if (xhttp.readyState === 4) {
+        if (xhttp.status == 200) resolve(xhttp.responseText)
+        else return reject(url_api);
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```